### ISSUE TITLE
Feature request: modify info.plist file after linking

### ISSUE THREAD

**Issue by xing-zheng** [rnpm/rnpm#132]
After linking the react-native-plugin, In some case, we should modify info.plist to add config, such as permission requirement or LSApplicationQueriesSchemes.

It will be perfect if we can specify these config in the package.json file like:

```javascript
"rnpm": {
    "ios": {
      "info.plist": [
        {key: "LSApplicationQueriesSchemes", value: ["facebook", "twitter"]}
      ]
    }
  }
```

**Comment by grabbou** [comment]
That sounds good - might have time to work on that later this week! Happy to accept a PR - if anyone wants to start working on that, please let me know and I'll be more than happy to introduce you to the unknowns of the codebase and how it all works in general!

**Comment by lostintangent** [comment]
If you end up doing this, could it be integrated with the new params feature in the link plugin? It would be really useful to add `Info.plist` entries whose value was actually source from user input.

**Comment by xing-zheng** [comment]
Sometimes, we should add keys to the info.plist, If we can do this by rnpm link params, It will be perfect:
```
rnpm link react-native-xxxx --plist APIKey=xxxxxxxxx
```

**Comment by grabbou** [comment]
Yeah, I think on iOS we will default to persisting user answers to Info.plist (maybe at some point we get them in `values/strings.xml` on Android as well)

**Comment by Kureev** [comment]
We will, but that's the next step.

**Comment by grabbou** [comment]
Would you guys mind linking a few Readme's where you advise users to add some settings to Info.plist & then optionally reference that in their native code (params)?

Might implement that before next Saturday when we are going to speak about `rnpm`, but would love to get some input first.

**Comment by lostintangent** [comment]
@grabbou If you take a look at the CodePush [iOS configuration](https://github.com/Microsoft/react-native-code-push#plugin-configuration-ios) docs, after the user adds the header import and changes how they load the JS bundle, there is a paragraph explaining how to update th Info.plist to add your key. Ideally, we could define a static name of the plist entry, and specify that the value would come from a param poll question. That way, we could configure iOS like we do on Android currently.

**Comment by grabbou** [comment]
> As a total aside, it would be awesome if an RNPM module could declare any headers it needed imported to the AppDelegate.m file, which would be similar to the way the Android auto imports the package into the MainActivity.java file.

That's something I've been thinking about recently (maybe not 100% the same, but I think it's somehow similar), basically to allow modules declare some places where they want their code be inserted aka `spots`, so e.g you can append `[Intercom setupWithApp:@"${yourApiKey}" appId:@"${appId}"]` to `AppDelegate.m`. We could store these then the values in e.g. `Info.plist` (or put them inline).

**Comment by grabbou** [comment]
What I like about `code-push` approach is that you guys automatically check Info.plist for settings (which unfortunately is not super common I imagine) - that greatly simplifies things since the only thing we have to do is to make sure we persist answers correctly

**Comment by lostintangent** [comment]
@grabbou That sounds perfect. I'm excited to continue simplifying our setup :D

**Comment by grabbou** [comment]
I am on this feature in the upcoming week! Sorry for the delay, but been quite busy with other things, I am going to have two takes on that:
1) Params to `Info.plist` (@lostintangent - I think you can read them directly from within your code, by just querying mainBundle, right? We don't need to pass you anything extra as on Android)
2) Arbitrary values to `Info.plist` - These are not params, but things like query schemes or other things (mainly needed for e.g. in app linking on iOS9).

There will be probably some blockers with the `android.params` discussion we had in a separate issue, but hopefully an initial implementation puts some light and we can figure out a better way to handle them both.

**Comment by lostintangent** [comment]
@grabbou Yep, the CodePush plugin already reads from the main bundle's Info.plist at runtime, so if RNPM could write key/value pairs to it after linking, with the key being specified by the plugin (e.g. "CodePushDeploymentKey"), and the value coming from the user input, that would be perfect.

As soon as you have a good idea of what the package.json config will look like for this, I'll update our plugin to take advantage of it.

**Comment by Kureev** [comment]
Can I help you guys solving this issue?
